src/protocols/hpr_http_router.erl

```diff
@@ -0,0 +1,89 @@
+%%%-------------------------------------------------------------------
+%%% @author jonathanruttenberg
+%%% @copyright (C) 2022, <COMPANY>
+%%% @doc
+%%%
+%%% @end
+%%% Created : 21. Sep 2022 12:10 PM
+%%%-------------------------------------------------------------------
+-module(hpr_http_router).
+-author("jonathanruttenberg").
+
+-include("../grpc/autogen/server/config_pb.hrl").
+-include("../grpc/autogen/server/packet_router_pb.hrl").
+
+-include("hpr_roaming.hrl").
+
+-export([send/4]).
+
+-spec send(
+    Packet :: hpr_packet_up:packet(),
+    GatewayStream :: hpr_router_stream_manager:gateway_stream(),
+    Route :: hpr_route:route(),
+    RoutingInfo :: hpr_routing:routing_info()
+) -> ok | {error, any()}.
+send(PacketUp, GatewayStream, Route, RoutingInfo) ->
+    WorkerKey = worker_key_from(PacketUp, Route),
+    PacketType = hpr_routing:routing_info_type(RoutingInfo),
+    PubKeyBin = hpr_packet_up:gateway(PacketUp),
+    Protocol = protocol_from(Route),
+
+    %%    start worker
+    case
+        hpr_http_sup:maybe_start_worker(
+            WorkerKey,
+            #{protocol => Protocol}
+        )
+    of
+        {error, worker_not_started, _} = Err ->
+            lager:error(
+                [{packet_type, PacketType}],
+                "failed to start http connector for ~p: ~p",
+                [hpr_utils:gateway_name(PubKeyBin), Err]
+            ),
+            Err;
+        {ok, WorkerPid} ->
+            lager:debug(
+                [
+                    {packet_type, PacketType},
+                    {protocol, http}
+                ],
+                "~s: [routing_info: ~p]",
+                [PacketType, RoutingInfo]
+            ),
+            hpr_http_worker:handle_packet(
+                WorkerPid, PacketUp, erlang:system_time(millisecond), GatewayStream, RoutingInfo
+            )
+    end,
+
+    todo,
+    ok.
+
+-spec worker_key_from(hpr_packet_up:binary(), hpr_route:route()) -> hpr_http_sup:worker_key().
```

Comment:
`hpr_packet_up:binary()` -> `hpr_packet_up:packet()`?

Reply:
Yes, that looks weird. I will change it.